Swift/match.playground/Contents.swift

```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+let testString = "one 1 two 2 three 3"
+
+let pattern = "[0-9]" // Matching numbers
+let nsString = testString as NSString // Cating to NSString to use special methods
+
+// Trying to initialize regex
+do {
+    let regex = try NSRegularExpression(pattern: pattern, options: [])
+    let results = regex.matches(in: testString, options: [], range: NSMakeRange(0, testString.characters.count)) // Matches
```

Comment:
You must use testString.utf16.count here. Because NSRegularExpression works with NSString, which are utf16 encoded.